### DISCUSSION TITLE
What partern are you using for examples dir structure?

### DISCUSSION THREAD

**Discussion by HosMercury** [maxcountryman/axum-login#182] [Q&A] [answered]
I know I could use any pattern, but I'm curious about the pattern ( dir structure ) in the examples!?

Web directory ( which resembles controllers aka handlers ) afaik .

I was using MVC and I know i could still .

But your pattern from where ? Or what is its name ?

Is it popular through rust or other lang?

**Comment by maxcountryman** [comment, marked as answer]
The examples are structured loosely to resemble [this project](https://github.com/codahale/yellhole). In my own projects, I use a similar structure and many similar patterns (especially regarding so called "services"). More generally, the core service pattern in the linked project is somewhat like the notion of a ["service layer"](https://martinfowler.com/eaaCatalog/serviceLayer.html); this fits into a larger design philosophy known as "Domain Driven Design" or DDD.
